src/services/transcript.ts

```diff
@@ -70,9 +90,65 @@ const categorizeItems = async (memberships: MembershipHistory[], jobs: StudentEm
 
   groupedMembershipHistory.activities = memberships;
 
-  groupedMembershipHistory.experiences.sort(
-    (a, b) => getExperienceEndDate(b) - getExperienceEndDate(a),
-  );
+  // Sort experiences by experience name then by end date
+  groupedMembershipHistory.experiences.sort((a, b) => {
+    const nameComparison = getName(a).localeCompare(getName(b));
+    return nameComparison !== 0
+      ? nameComparison
+      : getExperienceEndDate(b) - getExperienceEndDate(a);
+  });
+
+  let GroupByTitle = Object.entries(
+    Object.groupBy(jobs, (job: StudentEmployment) => job.Job_Title),
+  ).map(([title, job]) => ({
+    Job_Title: title,
+    job,
+    latestDate: '',
+  }));
+
+  // This code is has a lot of ? and ! opeators because typescript can't analyze the
+  // loop to recognize that the array indexing is safe.
+  for (let i = 0; i < GroupByTitle.length; i++) {
+    let maxDate: string = '';
+    let tempJob = GroupByTitle[i].job;
+    let numJobs = 0;
+    if (tempJob) {
+      numJobs = tempJob.length;
+    }
+    for (let j = 0; j < numJobs; j++) {
+      let tempVal = GroupByTitle[i];
+      let value = tempVal!.job?.[j].Job_End_Date;
+      if (maxDate! < value!) {
+        maxDate = value!;
+      }
+    }
+    GroupByTitle[i].latestDate = maxDate;
+  }
+
+  // Sorting the grouped job by the latest end date
+  GroupByTitle.sort((a, b) => {
+    return getLatestEndDate(b)! - getLatestEndDate(a)!;
+  });
```

Comment:
Return this new GroupByTitle in groupedMembershipHistory, NOT the de-grouped version below. (Why do all the work to combine each job into one entry with multiple times, and then not use it?)